File: backend/app/src/main/java/backend/model/game/GameResponse.java
package backend.model.game;

import backend.model.board.BoardResponse;
import backend.model.player.PlayerResponse;

// GameResponse contains public fields returned to the user in the API response
public class GameResponse {
    // Unique game id
    public final String id;

    // Game host
    public PlayerResponse host;

    // Game opponent
    public PlayerResponse opponent;

    // Board
    public final BoardResponse board;

    public GameResponse(Game game) {
        this.id = game.id;
        this.host = new PlayerResponse(game.host);
        if (game.opponent != null)
            this.opponent = new PlayerResponse(game.opponent);
        this.board = new BoardResponse(game.board);
    }
}

File: frontend/src/ts/lib/model/GameResponse.ts
import BoardResponse from "./BoardResponse.js";
import PlayerResponse from "./PlayerResponse.js";

export default class GameResponse {
  public id: string;

  public board: BoardResponse;

  public host: PlayerResponse;

  public opponent?: PlayerResponse;

  constructor(id: string, host: PlayerResponse, opponent?: PlayerResponse) {
    this.id = id;
    this.board = new BoardResponse();
    this.host = host;
    this.opponent = opponent;
  }
}
